File: node/test/unit/domain/Frete.test.ts
import Dimensao from "../../../src/domain/Dimensao"
import Frete from "../../../src/domain/Frete"
import Peso from "../../../src/domain/Peso"
import Produto from "../../../src/domain/Produto"


test("Deve calcular o valor do frete com base nas dimensões (altura, largura e profundidade em cm) e o peso dos produtos (em kg)", () => {
    const frete = new Frete()
    const camera = new Produto("2323-AFSRDHDKDL-456", "Camera",100, new Dimensao(20,15,10), new Peso(1))
    const guitarra = new Produto("8965-ASDFGHHJK-785", "Guitarra",100, new Dimensao(100,30,10), new Peso(3))
    frete.addProdutos(camera)
    frete.addProdutos(guitarra)

    expect(39.99).toEqual(frete.getValor())
})

test("Deve retornar o preço mínimo de frete caso ele seja superior ao valor calculado" , () => {
    const frete = new Frete()
    const camera = new Produto("2323-AFSRDHDKDL-456", "Camera",100, new Dimensao(20,15,10), new Peso(1))
    frete.addProdutos(camera)

    expect(10.00).toEqual(frete.getValor())
})

File: golang/src/domain/frete_test.go
package domain

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestFrete_GetValorQuandoValorFreteMenorQue10(t *testing.T) {
	camera := Produto{
		Codigo:    "2323-AFSRDHDKDL-456",
		Descricao: "Camera",
		Dimensao:  Dimensao{20, 15, 10},
		Peso:      Peso{1},
	}
	frete := Frete{}
	frete.AddProdutos(camera, 1)

	assert.Equal(t, 10.00, frete.GetValor())
}

func TestFrete_GetValorQuandoValorFreteMaiorQue10(t *testing.T) {
	camera := Produto{
		Codigo:    "2323-AFSRDHDKDL-456",
		Descricao: "Camera",
		Dimensao:  Dimensao{20, 15, 10},
		Peso:      Peso{1},
	}
	guitarra := Produto{
		Codigo:    "465454LKHLKHIOGK-789",
		Descricao: "Guitarra",
		Dimensao:  Dimensao{100, 30, 10},
		Peso:      Peso{3},
	}
	frete := Frete{}
	frete.AddProdutos(camera, 1)
	frete.AddProdutos(guitarra, 1)

	assert.Equal(t, 39.99, frete.GetValor())
}
